src/ZipFile.jl

```diff
@@ -513,8 +513,15 @@ end
 "advance io position state and crc32 checksum, checking it at eof"
 function update_reader!(f::ReadableFile, data::Array{UInt8})
     f._zpos = position(f._io) - f._datapos
-    f._pos += length(data)
-    f._currentcrc32 = Zlib.crc32(data, f._currentcrc32)
+    datalen = length(data)
+    f._pos += datalen
+    chunk_size = if Sys.WORD_SIZE > 32 2^31 else datalen end 
+    start = 1
+    while datalen > 0
```

Comment:
It seems `f._currentcrc32` will be uninitialized if `datalen == 0`, and then we're trying to use it below if `eof(f)`.

Reply:
`f._currentcrc32` should have already been initialized before this function is called. Please refer to line [https://github.com/fhs/ZipFile.jl/blob/8b2bbf3cc334228489a34cf911754c83309cba78/src/ZipFile.jl#L517](https://github.com/fhs/ZipFile.jl/blob/8b2bbf3cc334228489a34cf911754c83309cba78/src/ZipFile.jl#L517)

Reply:
That is the old code. After your change, that line is gone.

Reply:
Please note `f._currentcrc332` is used as an input in the previous version, which means it has been initialized already by that point. So it should be safe to use it in the new version of the code.

Reply:
I see. So, we're not making any new assumption about it being initialized or not.